src/AppInstallerCLITests/GroupPolicy.cpp

```diff
@@ -60,6 +60,50 @@ TEST_CASE("GroupPolicy_UpdateInterval", "[groupPolicy]")
     }
 }
 
+
+TEST_CASE("GroupPolicy_UpdateInterval_OldName", "[groupPolicy]")
+{
+    auto policiesKey = RegCreateVolatileTestRoot();
+
+    SECTION("New name shadows old")
+    {
+        SECTION("When old is valid")
+        {
+            SetRegistryValue(policiesKey.get(), SourceUpdateIntervalPolicyOldValueName, 3);
+        }
+        SECTION("When old is invalid")
+        {
+            SetRegistryValue(policiesKey.get(), SourceUpdateIntervalPolicyOldValueName, L"Invalid type");
+        }
+
+        SetRegistryValue(policiesKey.get(), SourceUpdateIntervalPolicyValueName, 1);
+        GroupPolicy groupPolicy{ policiesKey.get() };
+
+        auto policy = groupPolicy.GetValue<ValuePolicy::SourceAutoUpdateIntervalInMinutes>();
+        REQUIRE(policy.has_value());
+        REQUIRE(*policy == 1);
+    }
+
+    SECTION("Fallback to old name")
+    {
+        SECTION("When new name has invalid data")

```

Comment:
>When new name has invalid data [](http://example.com/codeflow?start=17&length=30)

if new name has invalid data, should we stop falling back?

Reply:
Agreed, the fallback should only happen when the new name is not present, not if it is bad data.  I think that may require additional or different registry API usage than what we have now (as I suspect that this behavior was based on how the templated function works).

Reply:
Updated to not fallback on invalid data.